### ISSUE TITLE
Record-destructuring in the REPL works wrong

### ISSUE THREAD

**Issue by mikeyhew** [gluon-lang/gluon#550]
Sorry if this has already been reported. I looked through the open issues, but couldn't see any issues about it.

Here is an example:

```
> let record = {foo = "foo", bar = "bar"}
> let {bar, foo} = record
> bar
"foo"
> foo
"bar"
```

It looks like the REPL "forgets" the names of fields, and just assigns the first name in the list to the value of the first field in the record.

**Comment by Marwes** [comment]
Thanks for reporting, should be fairly easy to fix actually.

**Comment by mikeyhew** [comment]
Cool, I'll try to fix it then. This is probably easier than #499, so I'll do this first.

**Comment by Marwes** [comment]
There is a `get_field` function on the `Value` type which takes  a string. You should be able to then just take the field name from the pattern as well.

**Comment by Marwes** [comment]
FYI, I will be away over the weekend so I will probably be unable to respond if you have any further questions until sometime Sunday.

**Comment by Marwes** [comment]
@mikeyhew Any luck with this yet? Any more questions?

**Comment by mikeyhew** [comment]
@Marwes things are going pretty well, actually. I fixed the record-destructuring bug and I will open a PR as soon as I finish writing a test case for it.

**Comment by Marwes** [comment]
Closed by #567
